Project.toml

```diff
@@ -24,6 +24,7 @@ GPUArraysCore = "46192b85-c4d5-4398-a991-12ede77f4527"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 LinearSolve = "7ed4a6bd-45f5-4d41-b270-4a48e9bafcae"
 Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
+Mooncake = "da2b9cff-9c12-43a0-ae48-6db2b0edb7d6"
```

Comment:
I'd be very happy to add it as an optional dep. I'm not 100% clear how I would do so in the current design though, because I've had to insert code directly into existing functions, rather than adding methods. e.g. in `adjointdiffcache`. Do you have a view on how we should make it optional?

> It also has ChainRules fallbacks right?

What exactly do you mean by this? We largely don't use ChainRules because most of the rules are only useful if you don't support mutation properly. We do permit users to say "please use this rrule" though, via [`@from_rrule`](https://compintell.github.io/Mooncake.jl/dev/utilities/tools_for_rules/#Using-ChainRules.jl) .

Reply:
> I'd be very happy to add it as an optional dep. I'm not 100% clear how I would do so in the current design though, because I've had to insert code directly into existing functions, rather than adding methods. e.g. in adjointdiffcache. Do you have a view on how we should make it optional?

Reply:
This isn't correct though? All of the new code is already in new functions?

https://github.com/SciML/SciMLSensitivity.jl/pull/1152/files#diff-1a15b4b5711133c125548ef7f1ca88f761bb124cffc8bfde8c13336968aaccd6R466-R471

https://github.com/SciML/SciMLSensitivity.jl/pull/1152/files#diff-1a15b4b5711133c125548ef7f1ca88f761bb124cffc8bfde8c13336968aaccd6R534-R541

If I'm not mistaken, those are the only two calls and both are already isolated. Since it's internal and you know the first is a tuple, you can just do:

```julia
function mooncake_run_ad(paramjac_config, y, p, t, λ)
   error("MooncakeVJP requires Mooncake.jl is loaded. Install the package and do `using Mooncake` to use this functionality")
end
```

and then just change the dispatch in the extension to `function mooncake_run_ad(paramjac_config::Tuple, y, p, t, λ)`?

Reply:
Mooncake is now a weakdep -- @ChrisRackauckas if you get approve CI to run when you get a minute.